### PR TITLE
Update my namespaces to work with inbound repos

### PR DESCRIPTION
**Pull request by ZitaNemeckova**
Fixes https://github.com/ansible/galaxy_ng/issues/290

Steps to test:
- if https://github.com/ansible/galaxy_ng/pull/272 not merged create an inbound repo `inbound-[namespace.name]` 
```python
>>> from pulp_ansible.app.models import AnsibleRepository, AnsibleDistribution
>>> repo = AnsibleRepository.objects.create(name='inbound-[namespace.name]')
>>> AnsibleDistribution.objects.create(name='inbound-[namespace.name]', base_path='inbound-[namespace.name]', repository=repo)
<AnsibleDistribution: automation-hub>
>>> # Press <CTRL+D> to exit.
```

   else create a namespace
- upload collection
- `/content/inbound-[namespace.name]/v3/artifacts/collection` is called instead of `automation-hub/v3/artifacts/collection`

*Add a new view where the user can copy the repository URL for the purpose of configuring the ansible-galaxy cli:*

This shows only in My Namespaces.

Screenshot 2020-08-03 at 16 33 35

### CODE REVIEW

File: src/containers/namespace-detail/namespace-detail.tsx [hunk 1]
```diff
@@ -111,14 +111,20 @@ export class NamespaceDetail extends React.Component<IProps, IState> {
...
 
     const tab = params['tab'] || 'collections';
 
     if (namespace.resources) {
       tabs.push('Resources');
     }
 
+    const repositoryUrl =
+      window.location.href.split('/ui')[0].replace('8002', '5001') +
+      '/api/automation-hub/content/inbound-' +
```
Comment (newswangerd) [inline src/containers/namespace-detail/namespace-detail.tsx:124]: Please use the API base path for the URL because this can change. I think you can get it with `CollectionAPI.apiBaseUrl`

https://github.com/ansible/ansible-hub-ui/blob/stable/src/api/base.ts#L7

Comment (ZitaNemeckova) [reply src/containers/namespace-detail/namespace-detail.tsx]: @newswangerd Did you mean to replace `/api/automation-hub/` by `CollectionAPI.apiBaseUrl`? How can I check that user owns the namespace? Or is it enough that the user belongs to same group as a namespace?

Comment (newswangerd) [reply src/containers/namespace-detail/namespace-detail.tsx]: `CollectionAPI.apiBaseUrl` should work.

You can just hide it if `showControls` is false. https://github.com/ansible/ansible-hub-ui/blob/master/src/containers/namespace-detail/namespace-detail.tsx#L55

Comment (ZitaNemeckova) [reply src/containers/namespace-detail/namespace-detail.tsx]: @newswangerd 
Hiding done in
https://github.com/ansible/ansible-hub-ui/pull/163/files#diff-155af35dcff67011de5d6c80469d78d3R117 
and `CollectionAPI.apiBaseUrl` used.

File: src/containers/namespace-detail/namespace-detail.tsx [hunk 2]
```diff
@@ -113,12 +114,22 @@ export class NamespaceDetail extends React.Component<IProps, IState> {
...
+      tabs.push('CLI Configuration');
+    }
     const tab = params['tab'] || 'collections';
 
     if (namespace.resources) {
       tabs.push('Resources');
     }
 
+    const repositoryUrl =
+      window.location.href.split('/ui')[0].replace('8002', '5001') +
```
Comment (newswangerd) [inline src/containers/namespace-detail/namespace-detail.tsx:127]: Don't split on `/ui`. The UI base path is configurable and cloud.redhat.com doesn't use that prefix for the UI.

I recommend using `window.location.origin`

Comment (newswangerd) [inline src/containers/namespace-detail/namespace-detail.tsx:127]: No need to replace the port. On the dev environment `8002` should redirect to `5001`.

File: src/containers/namespace-detail/namespace-detail.tsx [hunk 3]
```diff
@@ -113,12 +114,22 @@ export class NamespaceDetail extends React.Component<IProps, IState> {
...
+      CollectionAPI.apiBaseURL +
```
Comment (newswangerd) [inline src/containers/namespace-detail/namespace-detail.tsx:128]: On second thought, `CollectionAPI.apiBaseURL` might be a bad choice. We have the option to configure the UI to look for an api on a different host entirely, that's what `API_HOST` does here:

```
export class BaseAPI {
  apiBaseURL = API_HOST + API_BASE_PATH;
  apiPath: string;
  http: any;
```

`API_HOST` and `API_BASE_PATH` are both webpack constants that are defined here https://github.com/ansible/ansible-hub-ui/blob/master/config/standalone.dev.webpack.config.js#L8 if anyone decides to configure an `API_HOST` the URLs are going to break.

I would recommend is to check if `API_HOST` exists. If it does, use `API_HOST + API_BASE_PATH` for the first part of the URL. If it doesn't, use `window.location.origin + API_BASE_PATH`.

File: src/containers/namespace-detail/namespace-detail.tsx [hunk 4]
```diff
@@ -182,6 +193,17 @@ export class NamespaceDetail extends React.Component<IProps, IState> {
             ) : (
               this.renderResources(namespace)
             )}
+            {tab.toLowerCase() === 'cli configuration' ? (
+              <div>
+                <ClipboardCopy isReadOnly>{repositoryUrl}</ClipboardCopy>
+                <div>
+                  <b>Note:</b> Use this URL to configure ansible-galaxy to
```
Comment (newswangerd) [inline src/containers/namespace-detail/namespace-detail.tsx:200]: Reword this to "Use this URL to configure `ansible-galaxy` to upload collections to this namespace. More information on `ansible-galaxy` configurations can be found [here](https://docs.ansible.com/ansible/latest/galaxy/user_guide.html#configuring-the-ansible-galaxy-client)"

Comment (ZitaNemeckova) [reply src/containers/namespace-detail/namespace-detail.tsx:200]: Done :)

File: src/containers/namespace-detail/namespace-detail.tsx (hunk 3 above)
Comment (ZitaNemeckova) [reply src/containers/namespace-detail/namespace-detail.tsx]: Done :)

File: src/containers/namespace-detail/namespace-detail.tsx [hunk 5]
```diff
@@ -113,12 +114,23 @@ export class NamespaceDetail extends React.Component<IProps, IState> {
...
       tabs.push('Resources');
     }
 
+    const repositoryUrl = !!API_HOST
+      ? API_HOST
+      : window.location.origin +
+        API_BASE_PATH +
+        'content/inbound-' +
+        namespace.name +
+        '/v3/collections/';
```
Comment (newswangerd) [inline src/containers/namespace-detail/namespace-detail.tsx:132]: I think the client expects the URL to be the endpoint that returns this

```
{
    "available_versions": {
        "v3": "v3/"
    },
    "distro_base_path": "automation-hub"
}
```

So you can drop `v3/collections`. Make sure to leave the`/` at the end though.

Comment (ZitaNemeckova) [reply src/containers/namespace-detail/namespace-detail.tsx]: Fixed :)
